docs/how-to/extending/builtin.qmd

```diff
@@ -169,3 +172,60 @@ con.execute(cardinality("foo"))
 
 Here, Snowflake is informing us that the `ARRAY_SIZE` function does not accept
 strings as input.
+
+
+## Aggregate functions
+
+Let's the use the BigQuery backend to demonstrate how to access a BigQuery-only
+aggregate function: [`ARRAY_CONCAT_AGG`](https://cloud.google.com/bigquery/docs/reference/standard-sql/aggregate_functions#array_concat_agg).
+
+### BigQuery
+
+Connect to BigQuery:
+
+```{python}
+con = ibis.connect("bigquery://ibis-gbq")  # <1>
+
+con.set_database("bigquery-public-data.imdb")
+```
+
+1. Replace `ibis-gbq` with your GCP project name
+
+Now define the builtin aggregate function:
```

Comment:
A sentence or two about how the user should annotate the function signature would be useful. Perhaps describe what `array_concat_agg` does, and how the signature types below correspond to it?

Questions a user might ask:

- Why annotate only the return type and not the argument?
- Is the return type the aggregate? Or a column of the aggregate?
- What is `dt.Array[str]` (I know it's an array of strings, but users haven't had to interact with ibis datatypes in signatures anywhere else, so this is new to them).

I guess the general feedback here is "some more description of what's happening" would be nice.

Reply:
Alternatively, we could use a simpler backend and example which may reduce the complexity budget here. Perhaps wrapping duckdb's kurtosis function instead (https://duckdb.org/docs/sql/aggregates.html)?

This would also avoid the need for the biquery-specific callouts in the example around the connection string/use of `head`.

Reply:
Good points, I will adjust this to use DuckDB.

Reply:
Adjusted to use DuckDB and kurtosis, thanks for the feedback!